wrappers/kn/src/nativeMain/kotlin/zxingcpp/ImageView.kt

```diff
@@ -0,0 +1,72 @@
+package zxingcpp
+
+import cnames.structs.zxing_ImageView
+import kotlinx.cinterop.CPointer
+import kotlinx.cinterop.ExperimentalForeignApi
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.pin
+import zxingcpp.cinterop.*
+
+@OptIn(ExperimentalForeignApi::class)
+abstract class ImageView {
+	abstract val data: UByteArray
+	open val dataOffset: Int = 0
+	abstract val left: Int
+	abstract val top: Int
+	abstract val width: Int
+	abstract val height: Int
+	abstract val format: ImageFormat
+	abstract val rotation: Int
+	open val rowStride: Int = 0
+	open val pixStride: Int = 0
+
+	@OptIn(ExperimentalStdlibApi::class)
+	internal class ClosableCImageView(
+		data: UByteArray,
+		dataOffset: Int,
+		width: Int,
+		height: Int,
+		format: ImageFormat,
+		rotation: Int,
+		rowStride: Int,
+		pixStride: Int,
+	) : AutoCloseable {
+		private val pinnedData = data.pin()
+		val cValue: CPointer<zxing_ImageView>? =
+			zxing_ImageView_new(pinnedData.addressOf(dataOffset), width, height, format.rawValue, rowStride, pixStride).also {
+				zxing_ImageView_rotate(it, rotation)
+			}
+
+		override fun close() {
+			zxing_ImageView_delete(cValue)
+			pinnedData.unpin()
+		}
+
+		fun finalize() {
+			close()
+		}
+	}
+
+	internal val cValueWrapped: ClosableCImageView
+		get() = ClosableCImageView(data, dataOffset, width, height, format, rotation, rowStride, pixStride)
+}
+
+@OptIn(ExperimentalForeignApi::class)
+enum class ImageFormat(internal val rawValue: zxing_ImageFormat) {
+	None(zxing_ImageFormat_None),
+	Lum(zxing_ImageFormat_Lum),
+	RGB(zxing_ImageFormat_RGB),
+	BGR(zxing_ImageFormat_BGR),
+	RGBX(zxing_ImageFormat_RGBX),
+	XRGB(zxing_ImageFormat_XRGB),
+	BGRX(zxing_ImageFormat_BGRX),
+	XBGR(zxing_ImageFormat_XBGR)
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun zxing_ImageFormat.parseIntoImageFormat(): ImageFormat? =
+	ImageFormat.entries.firstOrNull { it.rawValue == this }
+
+@OptIn(ExperimentalForeignApi::class)
+fun Iterable<ImageFormat>.toValue(): zxing_ImageFormat =
```

Comment:
Those enum values can not be (meaningfully) ored together. That only makes sens for the `BarcodeFormat` enum.

Reply:
Oh, sorry, I forgot to remove this function, since it's simply copied from BarcodeFormat, I will remove this later.

Reply:
fixed